File: perl/fork/muti_proc.pl
#!/usr/bin/perl


use strict;
use warnings;
my $i=0;


while( $i < 5 ){
my $pid=fork();
my $var=100;
if($pid == 0){
		print "I'm child proc  ,my pid is  $$  parent pid is ".getppid().", var=".\$var."\n";
		sleep(5);
		exit(0);
}
	$i++;
}
wait();
while(1){}

File: c/fork/muti_proc.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>


int main()
{
		int i=0;
		while(i<5){
				pid_t pid = fork();
				if(pid == 0){
						printf("this is child proc ,"
						"pid is %d,parent pid is %d i=%d\n",getpid(),getppid(),i);
						sleep(i);
						exit(0);
				}else {
						i++;
						printf("this is parent proc,pid is %d\n",getpid());
				}
		}
		wait();
		printf("all child proc is over!\n");
		while(10);
}

